examples/chip-tool-darwin/commands/clusters/WriteAttributeCommandBridge.h

```diff
@@ -0,0 +1,114 @@
+        ModelCommand::AddArguments();
+    }
+
+    WriteAttribute(chip::ClusterId clusterId)
+        : ModelCommand("write-by-id")
+        , mClusterId(clusterId)
+    {
+        AddArgument("attribute-id", 0, UINT32_MAX, &mAttributeId);
+        AddArgument("attribute-value", &mAttributeValue);
+        AddArgument("timedInteractionTimeoutMs", 0, UINT16_MAX, &mTimedInteractionTimeoutMs);
+        ModelCommand::AddArguments();
+    }
+
+    WriteAttribute(const char * _Nonnull attributeName)
+        : ModelCommand("write")
+    {
+        AddArgument("timedInteractionTimeoutMs", 0, UINT16_MAX, &mTimedInteractionTimeoutMs);
+    }
+
+    ~WriteAttribute() {}
+
+    CHIP_ERROR SendCommand(CHIPDevice * _Nullable device, chip::EndpointId endpointId) override
+    {
+        chip::TLV::TLVWriter writer;
+        chip::TLV::TLVReader reader;
+
+        mData = static_cast<uint8_t *>(chip::Platform::MemoryCalloc(sizeof(uint8_t), mDataMaxLen));
+        VerifyOrReturnError(mData != nullptr, CHIP_ERROR_NO_MEMORY);
+
+        writer.Init(mData, mDataMaxLen);
+
+        mAttributeValue.Encode(writer, chip::TLV::AnonymousTag());
+        reader.Init(mData, writer.GetLengthWritten());
+        reader.Next();
+
+        id value = NSObjectFromCHIPTLV(&reader);
+
+        return WriteAttribute::SendCommand(device, endpointId, mClusterId, mAttributeId, value);
+    }
+
+    CHIP_ERROR SendCommand(CHIPDevice * _Nullable device, chip::EndpointId endpointId, chip::ClusterId clusterId,
+        chip::AttributeId attributeId, id _Nonnull value)
+    {
+        NSDictionary * valueDict = @{@"type" : @"Structure", @"value" : @[]};
+        dispatch_queue_t callbackQueue = dispatch_queue_create("com.chip.command", DISPATCH_QUEUE_SERIAL);
+        if (value != nullptr) {
+            valueDict = (NSDictionary *) value;
+        }
+        [device
+            writeAttributeWithEndpointId:[NSNumber numberWithUnsignedShort:endpointId]
+                               clusterId:[NSNumber numberWithUnsignedInteger:clusterId]
+                             attributeId:[NSNumber numberWithUnsignedInteger:attributeId]
+                                   value:valueDict
+                       timedWriteTimeout:mTimedInteractionTimeoutMs.HasValue()
+                           ? [NSNumber numberWithUnsignedShort:mTimedInteractionTimeoutMs.Value()]
+                           : nil
+                             clientQueue:callbackQueue
+                              completion:^(NSArray<NSDictionary<NSString *, id> *> * _Nullable values, NSError * _Nullable error) {
+                                  CHIP_ERROR chipError = [CHIPError errorToCHIPErrorCode:error];
+                                  if (chipError != CHIP_NO_ERROR) {
+                                      ChipLogError(chipTool, "Error: %s", chip::ErrorStr(chipError));
+                                  }
+                                  if (values) {
```

Comment:
I don't understand. When would `values` be non-nil here?  I guess the API has this, but we should get the API fixed, right?  @kpark-apple

Reply:
`values` would simply carry a single attribute path as in the write call. At this point, it's just there to use the same type for the callback block as in other APIs. Hence, it's fine to ignore the `values` unless the tool wants to use the path in the values instead of a closure to capture the requested path.